dspace-api/src/main/java/org/dspace/health/ItemCheck.java

```diff
@@ -35,153 +35,228 @@
+                oneStage.put("count", row.getValue());
+                stagesCountArray.put(oneStage);
             }
+            root.put("stagesCounts", stagesCountArray);
 
-            ret += String.format(
-                "\tWaiting for approval (workflow items): %d\n",
-                workflowItemService.countAll(context));
-
+            int waitingForApprovalCount = workflowItemService.countAll(context);
+            sb.append(String.format("\tWaiting for approval (workflow items): %d\n", waitingForApprovalCount));
+            root.put("waitingForApproval", waitingForApprovalCount);
         } catch (SQLException e) {
             error(e);
         }
 
         try {
-            ret += getObjectSizesInfo(context);
+            sb.append(getObjectSizesInfo(context, root));
             context.complete();
         } catch (SQLException e) {
             error(e);
         }
-        return ret;
+
+        this.setReportJson(root);
+        return sb.toString();
     }
 
 
-    public String getObjectSizesInfo(Context context) throws SQLException {
+    public String getObjectSizesInfo(Context context, JSONObject jo) throws SQLException {
         StringBuilder sb = new StringBuilder();
-        sb.append(String.format("Count %-14s: %s\n", "Bitstream",
-                                String.valueOf(bitstreamService.countTotal(context))));
-        sb.append(String.format("Count %-14s: %s\n", "Bundle",
-                                String.valueOf(bundleService.countTotal(context))));
-        sb.append(String.format("Count %-14s: %s\n", "Collection",
-                                String.valueOf(collectionService.countTotal(context))));
-        sb.append(String.format("Count %-14s: %s\n", "Community",
-                                String.valueOf(communityService.countTotal(context))));
-        sb.append(String.format("Count %-14s: %s\n", "MetadataValue",
-                                String.valueOf(metadataValueService.countTotal(context))));
-        sb.append(String.format("Count %-14s: %s\n", "EPerson",
-                                String.valueOf(ePersonService.countTotal(context))));
-        sb.append(String.format("Count %-14s: %s\n", "Item",
-                                String.valueOf(itemService.countTotal(context))));
-        sb.append(String.format("Count %-14s: %s\n", "Handle",
-                                String.valueOf(handleService.countTotal(context))));
-        sb.append(String.format("Count %-14s: %s\n", "Group",
-                                String.valueOf(groupService.countTotal(context))));
-        sb.append(String.format("Count %-14s: %s\n", "BasicWorkflowItem",
-                                String.valueOf(workflowItemService.countAll(context))));
-        sb.append(String.format("Count %-14s: %s\n", "WorkspaceItem",
-                                String.valueOf(workspaceItemService.countTotal(context))));
+
+        int bitstreamsCount = bitstreamService.countTotal(context);
```

Comment:
instead of repeating the code, cannot we have either generics or a simple orderedmap/list of what we want and iterate through this and do those countTotal,append,put only once?